middleware/common/unittest/source/test_transcode.cpp

```diff
@@ -100,7 +100,7 @@ namespace casual
          if( transcode::utf8::exist( "ISO-8859-15"))
          {
             const std::string source = { static_cast<std::string::value_type>(0xA4)};
-            const std::string expect( u8"€");
+            const std::string expect( "€");
             const std::string result = transcode::utf8::encode( source, "ISO-8859-15");
```

Comment:
Är detta bra? Bygger väl på att Eurosymbolen i expect råkar vara en UTF-8 encoded variant av symbolen. Dvs att källkoden editerats i en omgivning som har locale utf-8, och att kompilatorn accepterar denna byte-ström i den locale som gäller vid bygget. Borde man använda "hex-notation" i expected? Har inte läst på om det nya utf-8 stödet i C++20/23 ännu, men antar att det är det som gör at koden behövde ändras för at fungera med både C++17 och C++23...  Har köpt Josuttis "C++20 The complete Guide" (700 sidor om C++20, version daterad 2022-11-14) och ser att det finns ett avsnitt i den om ändringarna i utf-8 stödet.

Reply:
This is not that great... But  `u8` prefix creates a std::u8string, witch we don't have any knowledge off in our code base, yet. The source code encoding is in utf8, hence the euro-sign will be a utf8 encoded string, as far as I can understand. The whole u8 prefix is rather confusing, at least to me: https://stackoverflow.com/questions/23471935/how-are-u8-literals-supposed-to-work

It makes more sense now when it creates std::u8string.